discovery_engine_core/core/src/storage.rs

```diff
@@ -14,16 +14,101 @@
 
 use async_trait::async_trait;
 
-use crate::document::HistoricDocument;
+use crate::document::{self, HistoricDocument};
+
+use self::models::ApiDocumentView;
 
 pub mod sqlite;
 
 #[async_trait]
-pub trait Storage {
+pub trait Storage: FeedScope {
     type StorageError;
 
     async fn init_database(&self) -> Result<(), <Self as Storage>::StorageError>;
 
     async fn fetch_history(&self)
         -> Result<Vec<HistoricDocument>, <Self as Storage>::StorageError>;
+
+    fn feed(
+        &self,
+    ) -> &(dyn FeedScope<FeedScopeError = <Self as FeedScope>::FeedScopeError> + Send + Sync);
+}
+
+#[async_trait]
+pub trait FeedScope {
+    type FeedScopeError;
+
+    async fn close_feed_document(&self, document: document::Id)
```

Comment:
This are all operations on the Feed trait, we don't have to repeat feed in each of them